whois-smtp/src/main/java/net/ripe/db/whois/smtp/SmtpServer.java

```diff
@@ -0,0 +1,99 @@
+package net.ripe.db.whois.smtp;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import net.ripe.db.whois.common.ApplicationService;
+import net.ripe.db.whois.common.MaintenanceMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.net.InetSocketAddress;
+
+@Component
+public class SmtpServer implements ApplicationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SmtpServer.class);
+
+    private final SmtpServerChannelsRegistry smtpServerChannelsRegistry;
+    private final SmtpServerChannelInitializer smtpServerChannelInitializer;
+    private final MaintenanceMode maintenanceMode;
+    private final boolean smtpEnabled;
+    private final int smtpPort;
+
+    private Channel serverChannel;
+    private NioEventLoopGroup bossGroup;
+    private NioEventLoopGroup workerGroup;
+    int port;
+
+    @Autowired
+    public SmtpServer(
+            final SmtpServerChannelsRegistry smtpServerChannelsRegistry,
+            final SmtpServerChannelInitializer smtpServerChannelInitializer,
+            final MaintenanceMode maintenanceMode,
+            @Value("${mail.smtp.server.enabled:false}") final boolean smtpEnabled,
```

Comment:
Feel free to ignore.
We are creating a bunch of beans that maybe are not required.
How about using @Conditional annotation, as we do in elsewhere? and put that @Condition together with @Component, @Service...etc related to this module? so any bean will be created if mail.smtp.server.enabled is false

Reply:
maybe if you do this, you have an issue in the .xml when injecting dependencies...
because those dependencies are calculated at runtime, so when the .xml is read maybe are not yet there...and if the property is false will never be there.
So just ignore this

Reply:
Thanks Miguel, making the beans conditional indeed creates dependency problems. I prefer to leave the beans initialised but enable or disable the feature with a flag as it's simpler.